src/mumble/VolumeSliderWidgetAction.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "VolumeSliderWidgetAction.h"
+#include "VolumeAdjustment.h"
+
+#include <QSlider>
+#include <QToolTip>
+
+VolumeSliderWidgetAction::VolumeSliderWidgetAction(QObject *parent)
+	: QWidgetAction(parent), m_volumeSlider(createQtUniquePtr(new QSlider(Qt::Horizontal))) {
+	m_volumeSlider->setMinimum(-30);
+	m_volumeSlider->setMaximum(30);
+	m_volumeSlider->setAccessibleName(tr("Slider for volume adjustment"));
+	m_volumeSlider->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+
+	connect(m_volumeSlider.get(), &QSlider::valueChanged, this,
+			&VolumeSliderWidgetAction::on_VolumeSlider_valueChanged);
+	connect(m_volumeSlider.get(), &QSlider::sliderReleased, this,
+			&VolumeSliderWidgetAction::on_VolumeSlider_sliderReleased);
+
+	setDefaultWidget(m_volumeSlider.get());
+
+	m_volumeSlider->setProperty("mouseTracking", true);
+}
+
+void VolumeSliderWidgetAction::updateSliderValue(float value) {
+	int dbShift = VolumeAdjustment::toIntegerDBAdjustment(value);
+	m_volumeSlider->setValue(dbShift);
+	updateTooltip(dbShift);
+}
+
+void VolumeSliderWidgetAction::updateTooltip(int value) {
+	// ToolTip on hover
+
+	QString tooltipText;
+	if (value > 0) {
+		tooltipText = QString("%1: +%2 dB");
+	} else {
+		tooltipText = QString("%1: %2 dB");
+	}
```

Comment:
```suggestion
	QString tooltipText;
	toolTipText.sprintf("%%1: %+d dB", value);
```

Reply:
sprintf is [obsolete](https://doc.qt.io/qt-5/qstring-obsolete.html#sprintf). Are you sure?

Reply:
Yes I know, but I thought it was the only way to use explicit formatting options. However, it seems there is also [this](https://doc.qt.io/qt-5/qstring.html#arg-9) overload of `arg` which also allows formatting. If we can get the desired result with that, that'd of course be much better.
But looking at https://doc.qt.io/qt-5/qstring.html#argument-formats it appears as if the formatting is very limited and does not allow for a forced sign. So it seems we don't really have a choice other than using `sprintf` :eyes:

Reply:
Well, the other choice is to leave it as is or not to display an explicit '+' at all

Reply:
Or to make it really confusing to look at: Use the ``?`` operator inline as in ``.arg(value > 0 ? "+" : "")``

Reply:
> Or to make it really confusing to look at: Use the ? operator inline as in .arg(value > 0 ? "+" : "")

Right, that would also work. In the end this might even be the best choice at this point (as I don't like the code duplication (even if minor) in the current implementation :eyes: